include/aspect/material_model/interface.h

```diff
@@ -903,6 +903,31 @@ namespace aspect
     };
 
 
+    /**
+     * Additional output fields for phase properties to be added to
+     * the MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     */
+    template <int dim>
+    class PhasePropertyOutputs : public NamedAdditionalMaterialOutputs<dim>
```

Comment:
Actually I think this is general enough to be an addition to the base class `NamedAdditionalMaterialOutputs` and should be an independent pull request. The only modification to the base class is:
- Add a constructor to the base class that takes `property_names` and `n_points` and is implemented exactly like you do here
- Add a ` std::vector<std::vector<double> > property_values;` member to the base class
- Add a `get_nth_output` implementation to the base class that does exactly what you do here, except it also contains an `Assert` that the `property_values.size` is actually set to something > 0 (to make sure it was properly initialized).

Then you can simply use an object of type `NamedAdditionalMaterialOutputs` instead of `PhasePropertyOutputs`. And in the future the other derived classes could use the new functionality as well (they can keep their names if they want, but use the functionality you implemented here). I think this simplifies the implementation.

Reply:
Resolved by #3772